pyrealm/splash/splash.py

```diff
@@ -162,7 +167,7 @@ def estimate_initial_soil_moisture(
 
             # Loop over the calendar object, updating the soil_moisture array
             wn_day = wn_start
-            for day_idx in np.arange(366):
+            for day_idx in range(day1.days_in_year):
                 # Calculate aet, soil moisture and runoff:
```

Comment:
This is difficult - what if the first day is '1999-12-30' - that is then going to track forward 365 days in the data to '2000-12-28' because 2000 is a leap year. 

I know we've chatted about minor errors like this - it's also hard because the original implementation is not well set up for this. The original `main.py` assumes it is being fed chunks of data that are a year long and which start on January 1st - but doesn't enforce the length of the data. So - very hard to match the behaviour even if we wanted to.

The basic original behaviour though is to take a a time series (assumed 'XXXX-01-01' to 'XXXX-12-31') and then iterate until those two days have the same value. That's odd - I'd expect it to loop the extra day to make the "same" day match.

Either way, I'm not 100% sure what is better here, but this approach does definitely need documenting in the docstring and commenting to explain what the decision is (see also updating the comments on L155) and then testing to check that shifting the dates in the test benchmarks doesn't alter anything.

Reply:
I have fixed this with `pandas.Timestamp` and `pandas.DateOffset`. This can address non-January-1st start date and leap year, including starting with Feb 29th in a leap year.